### ISSUE TITLE
cmake: fail build in linux

### ISSUE THREAD

**Issue by sl1pkn07** [reicast/reicast-emulator#1574]
**Please Note: This form is the minimum required information for submitting bugs.**  
**Removing this form may lead to your issue being closed until it is completed.**

| Platform | Branch | Hash | CIDL |
| ------ | ------ | ---- | ---- |
| archlinux| master | ce90d43c344f19fdce915deef636ded933bc0d08 | GIT |

**_Description of the Issue_**

fail build in linux

**_Debugging Steps Tested_**

~~~
  cd build
  cmake ../reicast-emulator \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_INSTALL_PREFIX=/usr \
    -DUSE_QT=ON
~~~

**_Logs Gathered_**

```
[ 28%] Building CXX object CMakeFiles/reicast.dir/core/rend/gl4/gldraw.cpp.o
/usr/bin/c++  -DBUILD_COMPILER=0x30000002 -DCHD5_LZMA -DCMAKE_BUILD -DDEF_CONSOLE -DFEAT_AREC=0x40000001 -DFEAT_DSPREC=0x40000001 -DFEAT_HAS_SOFTREND=0 -DFEAT_SHREC=0x40000002 -DGLES -DHOST_CPU=0x20000004 -DHOST_OS=0x10000002 -DSUPPORT_X11 -DTARGET_NO_WEBUI -DUSE_EVDEV -D_7ZIP_ST -I/tmp/makepkg/sl1-reicast-emulator-git/src/build -I/tmp/makepkg/sl1-reicast-emulator-git/src/reicast-emulator -I/tmp/makepkg/sl1-reicast-emulator-git/src/reicast-emulator/core/deps -I/tmp/makepkg/sl1-reicast-emulator-git/src/reicast-emulator/core/deps/picotcp/include -I/tmp/makepkg/sl1-reicast-emulator-git/src/reicast-emulator/core/deps/picotcp/modules -I/tmp/makepkg/sl1-reicast-emulator-git/src/reicast-emulator/core/khronos -I/tmp/makepkg/sl1-reicast-emulator-git/src/reicast-emulator/core  -fno-operator-names -fpermissive -std=c++11  -msse4 -fopenmp -O3 -DNDEBUG   -o CMakeFiles/reicast.dir/core/rend/gl4/gldraw.cpp.o -c /tmp/makepkg/sl1-reicast-emulator-git/src/reicast-emulator/core/rend/gl4/gldraw.cpp
/tmp/makepkg/sl1-reicast-emulator-git/src/reicast-emulator/core/rend/gl4/gldraw.cpp: In function 'void CreateTextures(int, int)':
/tmp/makepkg/sl1-reicast-emulator-git/src/reicast-emulator/core/rend/gl4/gldraw.cpp:421:2: error: 'glTextureView' was not declared in this scope
  glTextureView(depthTexId, GL_TEXTURE_2D, stencilTexId, GL_DEPTH32F_STENCIL8, 0, 1, 0, 1);
  ^~~~~~~~~~~~~
/tmp/makepkg/sl1-reicast-emulator-git/src/reicast-emulator/core/rend/gl4/gldraw.cpp:421:2: note: suggested alternative: 'fbTextureId'
  glTextureView(depthTexId, GL_TEXTURE_2D, stencilTexId, GL_DEPTH32F_STENCIL8, 0, 1, 0, 1);
  ^~~~~~~~~~~~~
  fbTextureId
/tmp/makepkg/sl1-reicast-emulator-git/src/reicast-emulator/core/rend/gl4/gldraw.cpp: In function 'void gl4DrawStrips(GLuint)':
/tmp/makepkg/sl1-reicast-emulator-git/src/reicast-emulator/core/rend/gl4/gldraw.cpp:465:2: error: 'glProvokingVertex' was not declared in this scope
  glProvokingVertex(GL_LAST_VERTEX_CONVENTION);
  ^~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/reicast.dir/build.make:1015: CMakeFiles/reicast.dir/core/rend/gl4/gldraw.cpp.o] Error 1
make[2]: Leaving directory '/tmp/makepkg/sl1-reicast-emulator-git/src/build'
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/reicast.dir/all] Error 2
make[1]: Leaving directory '/tmp/makepkg/sl1-reicast-emulator-git/src/build'
make: *** [Makefile:87: all] Error 2
```

**_Screenshots_**

![Screenshot_20190427_205930](https://user-images.githubusercontent.com/462213/56853973-688d9780-692f-11e9-9db2-96a074e38838.png)

- [X] Place an X in the box to confirm you have filled in this form

**Comment by baka0815** [comment]
I can confirm this.

````
mkdir build
cd build
cmake ..
make
````

gives
````
[ 28%] Building CXX object CMakeFiles/reicast.dir/core/rend/gl4/gldraw.cpp.o
/usr/bin/c++  -DBUILD_COMPILER=0x30000002 -DCHD5_LZMA -DCMAKE_BUILD -DDEF_CONSOLE -DFEAT_AREC=0x40000001 -DFEAT_DSPREC=0x40000001 -DFEAT_HAS_SOFTREND=0 -DFEAT_SHREC=0x40000002 -DGLES -DHOST_CPU=0x20000004 -DHOST_OS=0x10000002 -DSUPPORT_X11 -DTARGET_NO_WEBUI -DUSE_EVDEV -D_7ZIP_ST -I/home/baka/svn/reicast-master/build -I/home/baka/svn/reicast-master -I/home/baka/svn/reicast-master/core/deps -I/home/baka/svn/reicast-master/core/deps/picotcp/include -I/home/baka/svn/reicast-master/core/deps/picotcp/modules -I/home/baka/svn/reicast-master/core/khronos -I/home/baka/svn/reicast-master/core  -fno-operator-names -fpermissive -std=c++11  -msse4 -fopenmp   -o CMakeFiles/reicast.dir/core/rend/gl4/gldraw.cpp.o -c /home/baka/svn/reicast-master/core/rend/gl4/gldraw.cpp
/home/baka/svn/reicast-master/core/rend/gl4/gldraw.cpp: In function 'void CreateTextures(int, int)':
/home/baka/svn/reicast-master/core/rend/gl4/gldraw.cpp:421:2: error: 'glTextureView' was not declared in this scope
  glTextureView(depthTexId, GL_TEXTURE_2D, stencilTexId, GL_DEPTH32F_STENCIL8, 0, 1, 0, 1);
  ^~~~~~~~~~~~~
/home/baka/svn/reicast-master/core/rend/gl4/gldraw.cpp:421:2: note: suggested alternative: 'fbTextureId'
  glTextureView(depthTexId, GL_TEXTURE_2D, stencilTexId, GL_DEPTH32F_STENCIL8, 0, 1, 0, 1);
  ^~~~~~~~~~~~~
  fbTextureId
/home/baka/svn/reicast-master/core/rend/gl4/gldraw.cpp: In function 'void gl4DrawStrips(GLuint)':
/home/baka/svn/reicast-master/core/rend/gl4/gldraw.cpp:465:2: error: 'glProvokingVertex' was not declared in this scope
  glProvokingVertex(GL_LAST_VERTEX_CONVENTION);
  ^~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/reicast.dir/build.make:1015: CMakeFiles/reicast.dir/core/rend/gl4/gldraw.cpp.o] Error 1
make[2]: Leaving directory '/home/baka/svn/reicast-master/build'
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/reicast.dir/all] Error 2
make[1]: Leaving directory '/home/baka/svn/reicast-master/build'
make: *** [Makefile:87: all] Error 2
````

**Comment by dmiller423** [comment]
yes,  will look at

**Comment by scribam** [comment]
Fixed with #1722, let me know if it is not the case.

**Comment by baka0815** [comment]
Trying with the current alpha branch cmake "works", reicast builds and can be run, but there are some errors in the compiled version:
- audio backends are missing (there are only "auto" and "none")
- the settings are not loaded (empty rom list, nothing configured, etc.) but my config exists at ``~/.config/reicast/emu.cfg``)

Using the compiled binary using make in ``reicast/linux`` works as expected.

**Comment by sl1pkn07** [comment]
running in any directory:

~~~
Warning: Unable to open the config file './emu.cfg' for reading (No such file or directory)
Creating new empty config file at './emu.cfg'
~~~

why search in `.` instead of `$HOME/.config/reicast`

?

**Comment by sl1pkn07** [comment]
seems https://github.com/reicast/reicast-emulator/blob/alpha/reicast/linux/Makefile#L30 uses a define `USES_HOMEDIR` wich is not presen in the Cmake build scripts

**Comment by sl1pkn07** [comment]
add under/near line 305 `set(_C_FLAGS "${_C_FLAGS} -D USES_HOMEDIR")`  into the file `reicast/cmake/config.cmake` do the trick

**Comment by scribam** [comment]
> audio backends are missing (there are only "auto" and "none")

Fixed in PR #1725

**Comment by sl1pkn07** [comment]
the main issue is fixed. closing it

greetings
